services/order.py

```diff
@@ -0,0 +1,36 @@
+from django.contrib.auth import get_user_model
+from django.db import transaction
+from django.db.models import QuerySet
+
+from db.models import Order, Ticket
+
+
+def create_order(
+        tickets: list[dict],
+        username: str,
+        date: str = None
+) -> Order:
+    with transaction.atomic():
+        order = Order.objects.create(
+            user=get_user_model().objects.get(username=username)
+        )
+        if date:
+            order.created_at = date
+            order.save()

```

Comment:
Don't Repeat Yourself: You are calling the `save()` method twice in this function. It would be more efficient to call it only once after all changes to the `order` object have been made. Here is the algorithm you can use: 1. Create an order object without saving it to the database. 2. If the `date` parameter is given, change the `created_at` attribute of the `order` object. 3. Save the `order` object to the database. 4. Create the tickets.